.travis.yml

```diff
@@ -16,27 +16,34 @@
 # specific language governing permissions and limitations
 # under the License.
 #
+sudo: true
 dist: xenial
 language: python
-python:
-  - "3.6"
 env:
   global:
-    - TRAVIS_CACHE=$HOME/.travis_cache/
+    - BUILD_ID=${TRAVIS_BUILD_ID}
+    - BRANCH_NAME=${TRAVIS_BRANCH}
   matrix:
-    - TOX_ENV=py27-backend_mysql-env_docker
-    - TOX_ENV=py27-backend_sqlite-env_docker
-    - TOX_ENV=py27-backend_postgres-env_docker
-    - TOX_ENV=py35-backend_mysql-env_docker PYTHON_VERSION=3
-    - TOX_ENV=py35-backend_sqlite-env_docker PYTHON_VERSION=3
-    - TOX_ENV=py35-backend_postgres-env_docker PYTHON_VERSION=3
-    - TOX_ENV=py27-backend_postgres-env_kubernetes KUBERNETES_VERSION=v1.9.0
-    - TOX_ENV=py35-backend_postgres-env_kubernetes KUBERNETES_VERSION=v1.13.0 PYTHON_VERSION=3
-
+    - BACKEND=mysql ENV=docker
+    - BACKEND=postgres ENV=docker
+    - BACKEND=sqlite ENV=docker
+    - BACKEND=postgres ENV=kubernetes KUBERNETES_VERSION=v1.9.0
+    - BACKEND=postgres ENV=kubernetes KUBERNETES_VERSION=v1.13.0
+python:
+  - '3.6'
+  - '3.5'
```

Comment:
This almost doubles  the number of test jobs we have - I'm not sure we should do that.

Reply:
No worries. It's mostly for testing. I will remove before we get to the point of merging.

Reply:
I made a comment / TODO in the code. 

Do you think we should stick with 3.5 or 3.6 is better choice? I'd vote for 3.6.

Reply:
Again, I'm confused. The tests will run inside of Docker, right?

Reply:
Correct. Tests run within Docker. Each python version has a separate image.

Reply:
I'd go for 3.6 as well. Can we remove 3.5?

Reply:
I'd be all for removing 3.5 but I believe a lot of users use 3.5 and I believe @ashb has a very strong opinion on that (it's the default python installed by Debian Stretch (Debian 9) - which is current "stable" Debian. 

However with https://wiki.debian.org/DebianBuster (release planned in less than a month) we will go straight to Python 3.7. So maybe at that time (and for 2.0.0 release in general) we should abandon 3.5 and switch to supporting both 3.6 and 3.7.

I think just switching to f-strings will be a major improvement for readability ;).

Maybe we should start a thread on DevList and vote on abandoning 3.5 ?

Reply:
Yes, we should be careful with this. For me, I try to support a Python version until it goes EOL.